### PR TITLE
Citations

### PR DESCRIPTION
**Pull request by ArisbethAg**
- [x] Added CitationDocuments migration and type.
- [x] Added inserts to Citation_documents table.
- [x] Added queries related to Citation_documents.
- [x] Connected front and back for createCitation.

### CODE REVIEW

File: api/source/controllers/citation.controller.ts [hunk 1]
```diff
@@ -0,0 +1,2 @@
+// (c) Tecnologico de Monterrey 2022, rights reserved.
+
```
Comment (iangg29) [inline api/source/controllers/citation.controller.ts:2]: Empty file?

File: api/source/database/migrations/20220909221517_citations.ts [hunk 2]
```diff
@@ -11,18 +11,21 @@ export async function up(knex: Knex): Promise<void> {
...
+
+      //Commented this since this part is not ready yet
+      //table.foreign("document_id").references(`${DOCUMENT_TABLE_NAME}.id`).deferrable("deferred");
+      // table
+      //   .foreign("document_id")
+      //   .references("id")
+      //   .inTable(DOCUMENT_TABLE_NAME)
+      //   .onUpdate("CASCADE")
+      //   .onDelete("CASCADE")
+      //   .deferrable("deferred");
```
Comment (iangg29) [inline api/source/database/migrations/20220909221517_citations.ts:28]: Remove commented code.

File: api/source/database/migrations/20220928050507_citation_documents.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  if (!(await knex.schema.hasTable(CITATION_DOCUMENTS_TABLE_NAME))) {
+    await knex.schema.createTable(CITATION_DOCUMENTS_TABLE_NAME, (table: CreateTableBuilder) => {
+      // COLUMNS
+      table.string("citation_id").notNullable;
+      table.string("document_type_id").notNullable;
+      // RELATIONSHIPS
+        table.foreign("citation_id").references(`${CITATION_TABLE_NAME}.id`).deferrable("deferred");
+        table.foreign("document_type_id").references(`${DOCUMENT_TYPE_TABLE_NAME}.id`).deferrable("deferred");
+    });
+  }
```
Comment (iangg29) [inline api/source/database/migrations/20220928050507_citation_documents.ts:17]: Implement `onUpdate` and `onDelete` methods. See other migration files as reference.

**Review by iangg29 — CHANGES_REQUESTED** [review]
LGTM 🤓
Just fix minor comments.

**Review by Naiztu — APPROVED** [review]
I like it

**Review by Naiztu — APPROVED** [review]
It looks good!

**Pull request merged**